tests/v2/validation/charts/gatekeeperallowednamespaces_test.go

```diff
@@ -0,0 +1,177 @@
+package charts
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	settings "github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/charts"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	namespaces "github.com/rancher/rancher/tests/framework/extensions/namespaces"
+	"github.com/rancher/rancher/tests/framework/pkg/environmentflag"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/stretchr/testify/assert"
+	require "github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	yaml "gopkg.in/yaml.v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type GateKeeperAllowedNamespacesTestSuite struct {
+	suite.Suite
+	client                        *rancher.Client
+	session                       *session.Session
+	project                       *management.Project
+	gatekeeperChartInstallOptions *charts.InstallOptions
+}
+
+func (n *GateKeeperAllowedNamespacesTestSuite) TearDownSuite() {
+	n.session.Cleanup()
+}
+
+func (n *GateKeeperAllowedNamespacesTestSuite) SetupSuite() {
+	testSession := session.NewSession(n.T())
+	n.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(n.T(), err)
+
+	n.client = client
+
+	// Get clusterName from config yaml
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(n.T(), clusterName, "Cluster name to install is not set")
+
+	// Get clusterID with clusterName
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(n.T(), err)
+
+	// get latest version of gatekeeper chart
+	latestGatekeeperVersion, err := client.Catalog.GetLatestChartVersion(charts.RancherGatekeeperName)
+	require.NoError(n.T(), err)
+
+	// Create project
+	projectConfig := &management.Project{
+		ClusterID: clusterID,
+		Name:      gatekeeperProjectName,
+	}
+	createdProject, err := client.Management.Project.Create(projectConfig)
+	require.NoError(n.T(), err)
+	require.Equal(n.T(), createdProject.Name, gatekeeperProjectName)
+	n.project = createdProject
+
+	n.gatekeeperChartInstallOptions = &charts.InstallOptions{
+
+		ClusterName: clusterName,
+		ClusterID:   clusterID,
+		Version:     latestGatekeeperVersion,
+		ProjectID:   createdProject.ID,
+	}
+
+}
+func (n *GateKeeperAllowedNamespacesTestSuite) TestGateKeeperAllowedNamespaces() {
+	subSession := n.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := n.client.WithSession(subSession)
+	require.NoError(n.T(), err)
+
+	if !client.Flags.GetValue(environmentflag.GatekeeperAllowedNamespaces) {
+		n.T().Skip()
+	}
+
+	n.T().Log("getting list of all namespaces")
+
+	sysNamespaces := settings.SystemNamespaces.Get()
+	sysNamespacesSlice := strings.Split(sysNamespaces, ",")
+	n.T().Log(sysNamespaces)
+
+	NSKinds := Kinds{
```

Comment:
Why don't put from line 92 to line 126 in a charts extension or in gatekeeperallowednamespaces.go where you set up a constraint. Where it will return the yaml in a string. So there is actually no need to write/read from a file.